tests/Kiota.Builder.Tests/Writers/Php/CodeMethodWriterTests.cs

```diff
@@ -1474,9 +1474,9 @@ public void WritesModelFactoryBodyForUnionModels()
         Assert.Contains("if ('#kiota.complexType1' === $mappingValue) {", result);
         Assert.Contains("$result->setComplexType1Value(new ComplexType1())", result);
         Assert.Contains("if ($parseNode->getStringValue() !== null) {", result);
-        Assert.Contains("$result->setStringValue($parseNode->getStringValue())", result);
+        Assert.Contains("$result->setStringValue($finalValue)", result);
```

Comment:
can we also validate the code declaring/assigning this final value is present please?